### ISSUE TITLE
Uncaught TypeError: Cannot read property 'getCharWidth' of null

### ISSUE THREAD

**Issue by kevinchiu** [atom-minimap/minimap#374]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.0.7
**System**: Mac OS X 10.10.4
**Thrown From**: [minimap](https://github.com/atom-minimap/minimap) package, v4.13.1

### Stack Trace

Uncaught TypeError: Cannot read property 'getCharWidth' of null

```
At /Users/kevin/.atom/packages/minimap/lib/minimap-element.coffee:431

TypeError: Cannot read property 'getCharWidth' of null
  at atom-text-editor-minimap.MinimapElement.measureHeightAndWidth (/Users/kevin/.atom/packages/minimap/lib/minimap-element.coffee:431:38)
  at atom-text-editor-minimap.MinimapElement.attachedCallback (/Users/kevin/.atom/packages/minimap/lib/minimap-element.coffee:91:6)
  at atom-pane.PaneElement.activeItemChanged (/Applications/Atom.app/Contents/Resources/app.asar/src/pane-element.js:150:24)
  at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:82:11)
  at Pane.module.exports.Pane.setActiveItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:242:22)
  at Pane.module.exports.Pane.activateItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:306:21)
  at tabs-tab.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/tabs/lib/tab-bar-view.js:210:24)
  at space-pen-ul.jQuery.event.dispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/jquery/dist/jquery.js:4435:9)
  at space-pen-ul.elemData.handle (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/jquery/dist/jquery.js:4121:28)

```

### Commands

```
     -5:06.2.0 command-palette:toggle (atom-text-editor.editor.mini.is-focused)
  2x -5:02.6.0 core:select-all (atom-text-editor.editor.mini.is-focused)
  2x -4:58.5.0 command-palette:toggle (atom-text-editor.editor.mini.is-focused)
     -4:55.5.0 core:move-down (atom-text-editor.editor.mini.is-focused)
     -4:54 core:move-up (atom-text-editor.editor.mini.is-focused)
     -4:53.7.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -4:53.7.0 window:install-shell-commands (atom-text-editor.editor.mini)
     -4:51.4.0 command-palette:toggle (atom-text-editor.editor.mini.is-focused)
  5x -4:49.4.0 core:move-down (atom-text-editor.editor.mini.is-focused)
     -4:45.6.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -4:45.6.0 settings-view:view-installed-packages (atom-text-editor.editor.mini)
     -1:35.8.0 command-palette:toggle (atom-text-editor.editor.is-focused)
     -1:34 core:select-all (atom-text-editor.editor.mini.is-focused)
     -1:29.9.0 core:move-down (atom-text-editor.editor.mini.is-focused)
     -1:29.6.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -1:29.6.0 settings-view:view-installed-themes (atom-text-editor.editor)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "minimap"
    ],
    "themes": [
      "one-dark-ui",
      "base16-tomorrow-dark-theme"
    ]
  }
}
```

### Installed Packages

```coffee
# User
auto-update-packages, v1.0.0
travis-ci-status, v0.16.0

# Dev
No dev packages
```

**Comment by abe33** [comment]
Do you happen to have updated the minimap prior to this error?

**Comment by averrin** [comment]
It happens when i Toggle mark in Vim-mode.

**Comment by abe33** [comment]
I believe it was fixed in 4.13.1, but if you perform an update from Atom the custom element won't be reloaded (you can't unregister a custom element, so after the update it's still the previous version that is used). Reloading atom should fix that, as it will now load the new version of the custom element.

**Comment by lukasjuhas** [comment]
This happened to me just now when I updated minimap to 4.13.2. Before updating, everything was fine.

**Comment by abe33** [comment]
@lukasjuhas Same explanation, on update the package is deactivated, but the custom element for the view can't be unregistered, so if there was faulty code in it it'll still be executed (until you reload Atom). 
In that case, I wasn't preventing measurement after the destruction of the view (which is now fixed in 4.13.2), so it could happen that the polling routine was triggered after the minimap reference have been freed. I hope this kind of error won't happen again now that all measurement is prevented after destruction.

**Comment by lukasjuhas** [comment]
@abe33 Make sense, thanks for that!

**Comment by rugk** [comment]
This happens to me when switching tabs.

**Comment by Tolmark12** [comment]
@rugk Same here.

**Comment by rceee** [comment]
Same here.

**Comment by Ahmed-90** [comment]
got this error too :scream:

**Comment by alanchavezgar** [comment]
Same here.

**Comment by rrsilver** [comment]
@abe33 Everything is fine after reloading. 

The Atom provide some hook to ask the user to reload the editor after some package update? Maybe is a path to follow.

**Comment by Ahmed-90** [comment]
it was fixed after reinstall / atom restart

**Comment by rugk** [comment]
I also don't see this error anymore.

**Comment by preco21** [comment]
Same issue!

I'm using Windows 10 and Atom latest

**Comment by AlanaMarie** [comment]
I have the same issue after updating the minimap.

**Comment by fungilation** [comment]
Same issue.

**Comment by abe33** [comment]
To all, I'll be closing this issue now, as the problem is really related to v4.13.1 and was fixed in v4.13.2, for the explanation as why you still get the error after the update, please see my comment above (https://github.com/atom-minimap/minimap/issues/374#issuecomment-135363998).

FWIW, I think the next major version of the minimap will probably have to include a change in the view so that we don't get this kind of issues in the future.

@rrsilver Not in my knowledge, but I may have missed some API, I'll check that, and maybe also take a look at the notification API to see if there's something to do with it on updates. However the real fix wil be to get rid of custom elements as it simply doesn't works with environment like Atom where code can be reloaded while running.
